primitives/src/lib.rs

```diff
@@ -63,9 +63,12 @@ pub const DEFAULT_COMMITTEE_SIZE: u32 = 4;
 pub const DEFAULT_BAN_MINIMAL_EXPECTED_PERFORMANCE: Perbill = Perbill::from_percent(0);
 pub const DEFAULT_BAN_SESSION_COUNT_THRESHOLD: SessionCount = 3;
 pub const DEFAULT_BAN_REASON_LENGTH: u32 = 300;
+
 pub const DEFAULT_CLEAN_SESSION_COUNTER_DELAY: SessionCount = 960;
 pub const DEFAULT_BAN_PERIOD: EraIndex = 10;
 
+pub const DEFAULT_FINALITY_VERSION: Version = 1;
```

Comment:
This is kind of controversial, if this should be bumped always or should it stay at 0 forever. Any thoughts @kostekIV ?

Reply:
There is an option to make a policy that this constant in the runtime should be always equal to the legacy version of finality in the host.

Reply:
the second option is better imo

Reply:
Ok, let me summon @timorl here -- shall this be `1`? Do we want this synchronized with the legacy finality version? How to guarantee that?

Reply:
I think this matters mostly for e2e tests, because it's only there where there can be a state that does not have this initialized...

Reply:
@DamianStraszak This definitely my reading of this value - only ever important for e2e tests and for setting up new chains. @timorl is there any particular reason that it was changed to `2` and then to `1`? The last change being https://github.com/Cardinal-Cryptography/aleph-node/pull/828

Reply:
@DamianStraszak If we want to not have to specify this in the chain spec, it is definitely possible - I would have to rework the test a bit but not that much. The downside being that we would not test the presence of a value for the finality version before the first change (but this is rather irrelevant as you have mentioned).

Reply:
I think there are generally two options here (this is for new chains, and/or e2e tests).
1) Set the initial value via chainspec, and leave `0` as default, always. This is a quite flexible approach, but the downside is that in all pipelines and whatnot we need to include a correct initial configuration in the chainspec.
2) Don't allow specifying this via chainspec, only specify this via `DEFAULT_FINALITY_VERSION` -- this is not quite flexible, but is simpler for pipelines and stuff.

I think I like 1) more. But let's hear @timorl and @kostekIV

Reply:
@timorl and @kostekIV any preference for either of the options mentioned by @DamianStraszak?

Reply:
Wait, I'm confused, I thought the default is only used for generating the chainspec in this PR, not in any other way, so it's already kinda 1) (which I also think is the better approach), isn't it? If my understanding is wrong, then I would support a solution like this anyway – make it chainspec-dependent (as in 1), but make the chainspec generation by default use the latest legacy version, avoiding the need for special setups in (most) tests.

Sorry for the late response, the way I read notifications on github made me miss this probably quite a couple of times. <_<

Reply:
Yes, prior to this PR there was no `GenesisConfig` value at all. The default finality version was used as follows https://github.com/Cardinal-Cryptography/aleph-node/blob/7c1d8220631e75b7830008d021f5f870c346ba80/pallets/aleph/src/lib.rs#L127. However, running e2e tests reveled that reading the value for the `FinalityVersion` immediately after the booting of the chain resulted in `None` values being returned. Hence, the addition of the default to `GenesisConfig`, which solved the issue.

Reply:
Ah, that explains why it was working for me...

Anyway, I would then definitely propose 1 + make the default chainspec config that is generated contain the default value (but not the default value being used as the default when the chainspec is unset, that should still be 0). Making chainspec generation produce different values for this setting can happen in a separate PR not to bloat this one (if possible).

Reply:
@DamianStraszak I have gone for `1.` as per the suggestions and made it part of this PR. Is now ready to re-read.